libs/platform/src/lib/components/form/date-picker/date-picker.component.ts

```diff
@@ -0,0 +1,261 @@
+/**
+ * @license
+ * SAP
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+import { NgControl, NgForm } from '@angular/forms';
+import { ChangeDetectorRef, ChangeDetectionStrategy } from '@angular/core';
+import { Component, EventEmitter, Input, Optional, Output, Self } from '@angular/core';
+import { ViewEncapsulation, ViewChild } from '@angular/core';
+import { CalendarType, CalendarYearGrid, DatePickerComponent as CoreDatePickerComponent } from '@fundamental-ngx/core';
+import { DaysOfWeek, FdCalendarView, FdDate, FdRangeDate, SpecialDayRule } from '@fundamental-ngx/core';
+import { Placement } from 'popper.js';
+import { FormFieldControl } from '../form-control';
+import { BaseInput } from '../base.input';
+
+/**
+ * Date-Picker implementation based on the
+ * https://github.com/SAP/fundamental-ngx/wiki/Platform:-Date-Picker-Component-Technical-Design
+ * documents.
+ *
+ * Core datePicker is wrapped here to work with platform form. Core datepicker properties
+ * and Event has been exposed in the same way.
+ */
+
+@Component({
+    selector: 'fdp-date-picker',
+    templateUrl: './date-picker.component.html',
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [{ provide: FormFieldControl, useExisting: DatePickerComponent, multi: true }]
+})
+export class DatePickerComponent extends BaseInput {
+    /**
+     * core datepicker as child
+     */
+    @ViewChild(CoreDatePickerComponent)
```

Comment:
What are the way for application developer to pass translated months and days?

Reply:
Translation getting applied to core calendar component. And Calendar has exposed CalendarI18n and CalendarI18nLabels services for translation. fdp-date-picker is component wrapping fd-date-picker which internally using core calendar. so have to use calendar services for translation.
please suggest if you have some approach in mind.

Reply:
This why I am racing this up. As I am not sure how we are going to work with core date picker. We have defined our 4 ways to translate and how how do you use service to translate to 50 languages?